skills/harden-telegram/tools/telegram_debug.py

```
@@ -1422,6 +1419,245 @@ def send_direct(text: str, chat_id: str | None = None) -> int:
+        token = _read_bot_token()
+    except RuntimeError as e:
+        print(f"send-reply failed: {e}", file=sys.stderr)
+        return 1
+
+    if not chat_id:
+        print("send-reply failed: --chat-id is required", file=sys.stderr)
+        return 1
+
+    url, data = build_reply_request(token, chat_id, text, reply_to)
+    try:
+        with urllib.request.urlopen(url, data=data, timeout=10) as resp:
```

Ruff (0.15.10)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)